test/e2e-tests.sh

```diff
@@ -106,7 +106,11 @@ function run_e2e_tests() {
   kubectl label namespace $2 istio-injection=enabled --overwrite
   local options=""
   (( EMIT_METRICS )) && options="-emitmetrics"
-  report_go_test -v -tags=e2e -count=1 ./test/$1 -dockerrepo gcr.io/knative-tests/test-images/$1 ${options}
+  if [ "$1" =  "e2e" ]; then
```

Comment:
I confess I'm not comfortable with this hardcoded approach, unless it proves to be way simpler and faster than properly and transparently making the tests able to run in parallel...

Reply:
report_go_test can determine whether it's running e2e or not based on input parameters, right?
In general, We can have an effective mode (say via a test flag). When it's in effective mode, test infra will be smart enough to run certain things in parallel. When it's not, everything is run sequentially. By default, we run effective mode.

Reply:
Yes. The first param is e2e or conformance, and we can check that, as I have done here. 

Running things in parallel can cause race conditions and add flakiness in the tests. This is not significantly faster(few minutes faster) than running them sequentially atleast for now as the tests other than autoscaler take max 1 min.

Reply:
`report_go_test` is test-type agnostic. Anyway, this would just move the hack to a different place (that's even worse, because the function is meant to be repo-agnostic).